### DISCUSSION TITLE
foreign key constraints vs. association_dependencies

### DISCUSSION THREAD

**Discussion by benhutton** [jeremyevans/sequel#1834] [General]
We're a little late to the game adding foreign key constraints. Our app is built with heavy use of the `association_dependencies` plugin. Is a foreign key constraint with a properly configured `on_delete` better than using the `association_dependencies` plugin? Is there any reason you'd still use `association_dependencies` even with the foreign key constraint `on_delete`?

**Comment by jeremyevans** [comment]
Assuming that the foreign key ON DELETE behavior is what you want, it's probably better than using association_dependencies.  For one, foreign key ON DELETE will handle all database changes, not just those that occur via the model layer.  association_dependencies can do this that foreign_key ON DELETE cannot handle, such as using the :destroy option to call the model level destroy (which can call before/after destroy hooks), so if you need that, that's a possible reason to continue using associaiton_dependencies.
